Lesson_12_hw/task_1.ts

```diff
@@ -0,0 +1,98 @@
+// 1. Создайте дженерик функцию getFirstElement, которая принимает массив элементов типа T, и возвращает первый элемент (типа T).
+
+function getFirstElement<T>(arr: T[]) {
+  return arr[0];
+}
+console.log(getFirstElement([1, 2, 3]));
+console.log(getFirstElement(["one", "two", "three"]));
+
+// 2. Создайте интерфейс Person, абстрактный класс Employee, который реализует интерфейс Person, и конкретные классы Manager и Developer.
+//   - Интерфейс Person должен содержать:
+//       Стринговые поля: name, surname, experienceYears
+//       Метод, возвращающий строку: getDetails().
+//   - Абстрактный класс Employee должен:
+//       Реализовывать интерфейс Person.
+//       Содержать защищенное поле: salary, не передающееся в конструктор (по дефолту 0)
+//       Содержать защищенный абстрактный метод: calculateSalary().,
+//         который считает зарплату и вызывается в конструкторе, и будет реализован в наследниках
+//   - Конкретные классы Manager и Developer должны:
+//       Наследоваться от Employee.
+//       Класс менеджер должен на конструкторе получать поле prefered, которое может быть только 'scrum' или 'kanban'
+//       Класс девелопер должен на конструкторе получать поле programmingLanguage, который может быть 'js', 'ts', 'java', 'python'
+//       Метод calculateSalary должен для менеджера устанавливать зарплату по формуле: количество лет опыта * 500
+//       Метод calculateSalary должен для девелопера устанавливать зарплату по формуле: количество лет опыта * 1000
+//       Реализовывать метод getDetails(), который должен выводить полную информацию об объекте вида:
+//         'My name is Elena TSovna, I am software developer with 6 years of experience in TypeScript and 6000$ salary' (пример для девелопера)
+
+interface Person {
+  name: string;
+  surname: string;
+  experienceYears: string;
+  getDetails(): string;
+}
+
+abstract class Employee implements Person {
+  name: string;
+  surname: string;
+  experienceYears: string;
+  abstract getDetails(): string;
+
+  protected abstract calculateSalary(): void;
+
+  protected salary: number = 0;
+
+  constructor(name: string, surname: string, experienceYears: string) {
```

Comment:
With this approach you do not need 35-37 lines
```suggestion
  constructor(public name: string, public surname: string, public experienceYears: string) {
```